wts/blueprints/aggregate.py

```diff
@@ -0,0 +1,115 @@
+
+    `GET /aggregate/user/user?filters=authz&filters=username`
+    """
+    # for `GET /aggregate/user/user`, flask sets endpoint to 'user/user'
+    endpoint = f"/{endpoint}"
+    if endpoint not in flask.current_app.config["AGGREGATE_ENDPOINT_ALLOWLIST"]:
+        raise NotFoundError(
+            "supplied endpoint is not configured in the Workspace Token Service aggregate endpoint allowlist"
+        )
+
+    filters = flask.request.args.getlist("filters")
+    parameters = flask.request.args.to_dict()
+    parameters.pop("filters", None)
+
+    if flask.request.headers.get("Authorization"):
+        authenticate(allow_access_token=True)
+        refresh_tokens = (
+            db.session.query(RefreshToken)
+            .filter_by(username=flask.g.user.username)
+            .filter(RefreshToken.expires > int(time.time()))
+            .order_by(RefreshToken.expires.asc())
+        )
+        #  if a user has multiple refresh tokens for the same commons, we want
+        #  the latest one to be used. see
+        #  https://stackoverflow.com/questions/39678672/is-a-python-dict-comprehension-always-last-wins-if-there-are-duplicate-keys
+        refresh_tokens = {
+            flask.current_app.config["OIDC"][rt.idp]["commons_hostname"]: rt
+            for rt in refresh_tokens
+        }
+        access_tokens = await asyncio.gather(
+            *[async_get_access_token(rt) for rt in refresh_tokens.values()]
+        )
+        request_info = [
+            (commons, endpoint, {"Authorization": f"Bearer {access_token}"})
+            if access_token
+            else (commons, None, None)
+            for commons, access_token in access_tokens
+        ]
+    else:
+        request_info = [
+            (commons, endpoint, {})
+            for commons in flask.current_app.config["COMMONS_HOSTNAMES"]
+        ]
+
+    commons_user_info = await asyncio.gather(
+        *[
+            make_request(commons, endpoint, headers, parameters, filters)
+            for commons, endpoint, headers in request_info
+        ]
+    )
+
+    return flask.jsonify({c: i for c, i in commons_user_info})
+
+
+async def make_request(commons_hostname, endpoint, headers, parameters, filters):
+    """
+    Make an asychronous request to `endpoint` on `commons_hostname`.
+    """
+    if endpoint is None:
+        return (commons_hostname, {})
+    endpoint_url = f"https://{commons_hostname}{endpoint}"
+
+    try:
+        async with httpx.AsyncClient() as client:
+            endpoint_response = await client.get(
+                endpoint_url, headers=headers, params=parameters
+            )
+            endpoint_response.raise_for_status()
+    except httpx.RequestError as e:
+        flask.current_app.logger.error("Failed to get response from %s.", e.request.url)
+        return (commons_hostname, {})
+    except httpx.HTTPStatusError as e:
+        flask.current_app.logger.error(
+            "Status code %s returned from %s",
```

Comment:
I'd prefer we consistently use either `.format` or f-strings rather than the %s syntax, I former are way more prevalent in our codebase.